### PR TITLE
added maxsize function

### PR DESCRIPTION
**Pull request by ColleagueRiley**
not sure if I did anything else but whatever have fun with this

### CODE REVIEW

File: silicon.h [hunk 1]
```diff
@@ -43,6 +43,7 @@
 
 #define NS_ENUM(type, name) type name; enum
 
+#define GL_SILENCE_DEPRECATION
```
Comment (EimaMei) [inline silicon.h]: GL_SILENCE_DEPRECATION mustn’t be pre-defined

File: silicon.h [hunk 2]
```diff
@@ -1552,10 +1553,12 @@ enum { /* classes */
     NS_WINDOW_STYLE_MASK_CODE,
     NS_STRING_FROM_CLASS_CODE,
     NS_STRING_IS_EQUAL_CODE,
+    NS_WINDOW_SET_MAX_SIZE_CODE,
+    NS_WINDOW_SET_MIN_SIZE_CODE,
```
Comment (EimaMei) [inline silicon.h]: Where’s NS_CODE_LEN from the latest commit?

File: silicon.h [hunk 3]
```diff
@@ -1552,10 +1553,12 @@ enum { /* classes */
...
 };
 
-void* SI_NS_CLASSES[36] = {NULL};
-void* SI_NS_FUNCTIONS[233];
+void* SI_NS_CLASSES[38] = {NULL};
+void* SI_NS_FUNCTIONS[236];
```
Comment (EimaMei) [inline silicon.h]: Should be SI_NS_CLASSES[NS_CODE_LEN]

File: silicon.h [hunk 4]
```diff
@@ -2128,6 +2133,20 @@ void NSWindow_setTitle(NSWindow* window, const char* title) {
...
+
+    return ((void (*)(id, SEL, NSSize))objc_msgSend)
+                (SI_NS_CLASSES[NS_WINDOW_CODE], func, size);
+}
+
+void NSWindow_setMinSize(NSWindow* window, NSSize size) {
+    void* func = SI_NS_FUNCTIONS[NS_WINDOW_SET_MIN_SIZE_CODE];
+
+    return ((void (*)(id, SEL, NSSize))objc_msgSend)
+                (NSAlloc(SI_NS_CLASSES[NS_WINDOW_CODE]), func, size);
```
Comment (EimaMei) [inline silicon.h]: What’s NSAlloc doing here?

**Review by EimaKve — CHANGES_REQUESTED** [review]

**Comment by ColleagueRiley** [comment]
should be fixed, although my version of silicon is still older so you should pull only the changes I made (unless there weren't many new changes between this)

**Review by EimaKve — APPROVED** [review]

**Pull request merged**